docs/developer-docs/latest/plugins/upload.md

```diff
@@ -123,20 +125,48 @@ To upload files into your application.
 </form>
 
 <script type="text/javascript">
-  const formElement = document.querySelector('form');
+  const form = document.querySelector('form');
 
-  formElement.addEventListener('submit', (e) => {
+  form.addEventListener('submit', async (e) => {
     e.preventDefault();
 
-    const request = new XMLHttpRequest();
-
-    request.open('POST', '/upload');
-
-    request.send(new FormData(formElement));
+    await fetch('/api/upload', {
+      method: 'post',
+      body: new FormData(e.target)
+    });
   });
 </script>
 ```
 
+:::
+
+::: tab Node.js
+
+### From Node.js
```

Comment:
```suggestion
### From Node.js
```

We try to avoid using headers inside of tabs. VuePress does not allow direct linking to any tab or content inside a tab.